packages/components/src/Avatar/src/Avatar.tsx

```diff
@@ -0,0 +1,254 @@
+        const character = name.charCodeAt(i);
+        const shift = i % 8;
+
+        hashCode ^= (character << shift) + (character >> (8 - shift));
+    }
+
+    return hashCode % maxNumberOfColor;
+}
+
+function getColorName(name: string) {
+    const variantToUse = `option${getColorIndex(name, 8) + 1}`;
+
+    return `decorative-${variantToUse}`;
+}
+
+function AvatarInitials(props: AvatarInitialsProps) {
+    const { 
+        name,
+        size: sizeValue
+    } = props;
+    
+    const size = useResponsiveValue(sizeValue) ?? "md";
+
+    const initials = useMemo(() => {
+        const cleanName = name.replace(/\s+/g, " ").trim();
+        const [firstName, lastName] = cleanName.split(" ");
+        const letters = firstName && lastName
+            ? `${firstName.charAt(0)}${lastName.charAt(0)}`
+            : firstName.charAt(0);
+
+        return size === "xs" && letters.length > 1 ? letters.charAt(0) : letters;
+    }, [name, size]);
+
+    return (
+        <Text
+            size={AvatarToTextSizeAdapter[size]}
+            className={styles["hop-Avatar__initials"]}
+        >
+            {initials}
+        </Text>
+    );
+}
+
+function Avatar(props: AvatarProps, ref: ForwardedRef<HTMLDivElement>) {
+    [props, ref] = useContextProps(props, ref, AvatarContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        "aria-label": ariaLabel,
+        className,
+        fallbackSrc,
+        imageProps,
+        isDisabled,
+        name,
+        size: sizeValue,
+        slot,
+        src,
+        style,
+        ...otherProps
+    } = ownProps;
+    
+    const { onError, ...otherImageProps } = imageProps ?? {};
+    const [imageUrl, handleImageError, imageFailed] = useImageFallback(src, fallbackSrc);
+
+    const size = useResponsiveValue(sizeValue) ?? "md";
+
+    let classNames = clsx(
+        className,
+        GlobalAvatarCssSelector,
+        cssModule(
+            styles,
+            "hop-Avatar",
+            size,
+            getColorName(name)
+        ),
+        stylingProps.className
+    );
+
+    const brokenImageClassNames = clsx(
+        className,
+        GlobalAvatarCssSelector,
+        cssModule(
+            styles,
+            "hop-Avatar",
+            "broken-image"
+        ),
+        stylingProps.className
+    );
+
+    const mergedStyles: CSSProperties = {
+        ...stylingProps.style,
+        ...style
+    };
+
+    const initialsContent = <AvatarInitials
+        {...otherProps}
+        name={name}
+        size={size}
+    />;
+
+    const commonProps = {
+        ...otherProps,
+        ariaLabel,
+        ref,
+        size,
+        style: mergedStyles
+    };
+
+    let content = initialsContent;
+    
+    if (src) {
+        if (imageFailed) {
+            if (fallbackSrc === undefined) {
+                return (
+                    <RichIconAvatarImage
+                        {...otherProps}
```

Comment:
other props are already in commonProps? no need to spread both

Reply:
well some of them that were in common props were not in otherProps such as size